packages/assistant-stream/src/core/utils/stream/LineDecoderStream.ts

```
@@ -0,0 +1,26 @@
+    super({
+      transform: (chunk, controller) => {
+        this.buffer += chunk;
+        const lines = this.buffer.split("\n");
+
+        // Process all complete lines
+        for (let i = 0; i < lines.length - 1; i++) {
+          controller.enqueue(lines[i]);
+        }
+
+        // Keep the last incomplete line in the buffer
+        this.buffer = lines[lines.length - 1]!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)